### DISCUSSION TITLE
How to check TPM/NOPM values in logs for db2 database in linux

### DISCUSSION THREAD

**Discussion by Prathyusha6** [TPC-Council/HammerDB#162] [Q&A] [answered]
Hi @sm-shaw ,

Details:
DB2 version: 11.5.4.0
Hammerdb: 3.3
Working on RHEL 7.6

While running creating db2 schema in hammerdb GUI, 
Tried this following command and am getting tpm number, but the GUI image from hammerdb is showing a different number always(tried with different number of users and warehouses as well).
 
`select total_app_commits+ total_app_rollbacks from sysibmadm.mon_db_summary`

am getting these two results at the same time, aren't these two supposed to give the same tpm ?

![V1W20_CLI_tpm](https://user-images.githubusercontent.com/12022780/97561787-b09ae800-1a06-11eb-8832-01a1345ee4d7.PNG)
![V1W20_GUI_tpm](https://user-images.githubusercontent.com/12022780/97561791-b2fd4200-1a06-11eb-91b8-0255d35aede6.PNG)

Also, can you please guide me where to get the logs with tpm/nopm details of the existing databases in linux (hammerdb and db2 running on the same machine)

Thanks,
Prathyusha

**Comment by sm-shaw** [comment]
The TPM in the GUI is a sample over the refresh rate by default 10 secs and then the calculation to show what this number would be over 1 minute.  The statement you have is the number of transactions since the instance started - so these are not expected to be the same.  By default when you select log output in the virtual user options on Linux it will go into the /tmp directory.

**Comment by Prathyusha6** [comment]
Thanks @sm-shaw.

Also, is there a way to tune the db2/db_settings to get uniform tpm rate without any dips as seen in the picture above? 

Thanks,
Prathyusha

**Comment by sm-shaw** [comment, marked as answer]
Yes of course, that should be your goal for consistent performance however this tuning is out of scope for HammerDB support and is more for Db2 performance tuning and monitoring. As a starting point I would recommend db2top/dmstop to identify where the database is spending most time in wait events https://datageek.blog/en/2017/03/07/comparing-dsmtop-and-db2top/

**Comment by sm-shaw** [comment]
Closed with no further update, Issue is for Db2 tuning rather than HammerDB.
